src/Operation/AsyncMap.php

```diff
@@ -41,6 +44,10 @@ final class AsyncMap extends AbstractOperation
      */
     public function __invoke(): Closure
     {
+        if (PHP_VERSION_ID >= Constant::PHP_VERSION_8) {
+            throw new RuntimeException('This operation is currently not supported on PHP 8.');
+        }
```

Comment:
this needs to be inside the class otherwise PHPStan fails when discovering the files and that error cannot be ignored. Won't make a difference for the end user